benchmarks/AiDotNet.PyTorchParity/pytorch/benchmark.py

```
@@ -0,0 +1,345 @@
+class ResourceMonitor:
+    def __init__(self, sample_seconds: float = 0.1) -> None:
+        self.sample_seconds = sample_seconds
+        self.process = psutil.Process(os.getpid())
+        self.cpu: list[float] = []
+        self.rss_mb: list[float] = []
+        self.gpu_util: list[float] = []
+        self.gpu_mem: list[float] = []
+        self._stop = threading.Event()
+        self._thread = threading.Thread(target=self._sample, daemon=True)
+
+    def __enter__(self) -> "ResourceMonitor":
```

Ruff (0.15.14)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)